src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +181,42 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        public bool AutoDetectScriptSuccess()
+        {
+            // Only check WinHttpDetectAutoProxyConfigUrl if auto detect is set.
+            if (!AutoDetect) return false;
+
+            bool doAutoDetect = false;
+            int currentTimeStamp = Environment.TickCount;
+
+            // Environment.TickCount will increment from zero to Int32.MaxValue for approximately 24.9 days, then jump to
+            // Int32.MinValue. When that happens, we may skip calling WinHttpDetectAutoProxyConfigUrl for one request which we
+            // otherwise will call. Since _previousTimeStampTicks will get updated in that request, subsequent requests in
+            // next 24.9 days period will behave correctly.
+            doAutoDetect = currentTimeStamp - _previousTimeStampTicks > TimerPeriodInMilliseconds;
+            _previousTimeStampTicks = currentTimeStamp;
```

Comment:
My previous comment seems to have disappeared... trying again...

This is wrong. As long as another request is made within the next two minutes, it's going to set _previousTimeStampTicks to the current time, such that the next request made within the next two minutes will also skip detection. This line needs to be moved to inside of the doAutoDetect if block, so that the time is only updated when we're going to refresh.

Reply:
I found out that `currentTimeStamp - _previousTimeStampTicks` will cause a overflow for the first request (when `_previousTimeStampTicks = int.MinValue`), thus we are skipping calling `WinHttpDetectAutoProxyConfigUrl` for the first request. 

```
            int a = 1;
            int b = Int32.MinValue;
            Console.WriteLine(a - b);

output = -2147483647.
```

In most of our test, we will only send one request, that's why there is no regression. (We should expect some).

After I correct this and test locally, the execution time is back to 375.699s. Seems this regression is inevitable in test.

Reply:
> Seems this regression is inevitable in test.

I don't think this is acceptable.  Even if in general this cost is only going to impact the first request made per SocketsHttpHandler, there are a variety of scenarios where today only one request is ever made per SocketsHttpHandler, e.g. ClientWebSocket, HttpWebRequest, etc.  We can't hit those with a 10x regression.